projects/core/utils/mask/create-number-mask.ts

```diff
@@ -115,7 +115,11 @@ export function tuiCreateNumberMask({
 function preventLeadingZeroes(mask: Array<string | RegExp>): Array<string | RegExp> {
     const firstDigitIndex = mask.indexOf(TUI_DIGIT_REGEXP);
 
-    if (firstDigitIndex !== -1 && mask[firstDigitIndex + 1] === TUI_DIGIT_REGEXP) {
+    if (
+        firstDigitIndex !== -1 &&
+        mask[firstDigitIndex] === TUI_DIGIT_REGEXP &&
```

Comment:
This is a useless check, we already know it's true because this is how we got `firstDigitIndex` in the first place. From what I understand, we check that there's a digit in the mask and there's another digit after it. You changed it to just see if there is a digit in the mask and the mask is not one symbol. Could you elaborate on this change? Now that it doesn't look for two digits in a row, could it have issues with decimal part separator?